src/test/java/nl/hsac/fitnesse/util/RandomDomainTest.java

```diff
@@ -0,0 +1,77 @@
+package nl.hsac.fitnesse.util;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests RandomUtil.
+ */
+public class RandomDomainTest {
+    private final RandomDomain domain = new RandomDomain();
+    private final RandomUtil random = new RandomUtil();
+
+    @Rule
+    public final ExpectedException exception = ExpectedException.none();
+
+    /**
+     * Tests tld randomization.
+     */
+    @Test
+    public void testRandomTld() {
+        for (int i = 0; i < 5000; i++) {
+            String result = domain.getRandomTld();
+            boolean resultExists = false;
+
+            for (RandomDomain.tlds tlds : RandomDomain.tlds.values()) {
+                if (tlds.name().equalsIgnoreCase(result)) {
+                    resultExists = true;
+                    break;
+                }
+            }
+            assertTrue("Got: " + result + " on test " + i, resultExists);
+        }
+    }
+
+    /**
+     * Tests domain generation.
+     */
+    @Test
+    public void testGetRandomDomain() {
+        for (int i = 0; i < 1000; i++) {
+            int length = random.random(100) + 1; //considering a randomizer will generate non-inclusive from 0
+            String result = domain.getRandomDomain(length);
+            assertTrue("Got: " + result + " on test " + i, result.length() <= 100);
+            assertTrue("Got: " + result + " on regex test " + i, result.matches("[abcdefghijklmnopqrstuvwxyz1234567890-]+"));
+        }
+    }
+
+    /**
+     * Tests top and second level domain construction.
+     */
+    @Test
+    public void testGenerateFullDomain() {
+        for (int i = 0; i < 1000; i++) {
+            int length = random.random(95) + 5; //5 to use as minimal length for a hostname in email
+            String result = RandomDomain.generateFullDomain(domain, length);
+            assertTrue("Got: " + result + " on test " + i, result.length() <= 100);
+            assertTrue("Got: " + result + " on regex test " + i, result.matches("[a-z0-9-]+" + "." + "[A-Z]+"));
+        }
+    }
+
+    /**
+     * Tests error message if requested domain length is less than 5.
+     */
+    @Test
+    public void testDomainLengthException() {
+        exception.expect(IllegalArgumentException.class);
```

Comment:
To my understanding this test method only ensures at least one of the method calls in it throws an exception.
Do you expect all of them to throw or only the last? Either write separate test methods for all, or only make the call that you expect to throw...